identity-iota/src/tangle/client.rs

```diff
@@ -74,23 +76,37 @@ impl Client {
   }
 
   /// Publishes an [`IotaDocument`] to the Tangle.
-  pub async fn publish_document(&self, document: &IotaDocument) -> Result<Receipt> {
-    self.publish_json(document.integration_index(), document).await
+  pub async fn publish_document(&self, document: &IotaDocument, compression: bool) -> Result<Receipt> {
+    self
+      .publish_json(document.integration_index(), document, compression)
+      .await
   }
 
   /// Publishes a [`DocumentDiff`] to the Tangle to form part of the diff chain for the integration
   /// chain message specified by the given [`MessageId`].
-  pub async fn publish_diff(&self, message_id: &MessageId, diff: &DocumentDiff) -> Result<Receipt> {
-    self.publish_json(&IotaDocument::diff_index(message_id)?, diff).await
+  pub async fn publish_diff(&self, message_id: &MessageId, diff: &DocumentDiff, compression: bool) -> Result<Receipt> {
+    self
+      .publish_json(&IotaDocument::diff_index(message_id)?, diff, compression)
+      .await
   }
 
-  /// Publishes arbitrary JSON data to the specified index on the Tangle.
-  pub async fn publish_json<T: ToJson>(&self, index: &str, data: &T) -> Result<Receipt> {
+  /// Compresses and Publishes arbitrary JSON data to the specified index on the Tangle.
+  pub async fn publish_json<T: ToJson>(&self, index: &str, data: &T, compression: bool) -> Result<Receipt> {
```

Comment:
If this feature: https://blog.rust-lang.org/inside-rust/2021/09/06/Splitting-const-generics.html#featureconst_generics_defaults lands before our 1.0 release, then we can use that to essentially set `compression = true` as a default, and then one has to write a bit of boiler plate in the rare cases where no compression is desired. For the time being it is probably fine to have `compression` as a parameter.

Reply:
The compression argument is removed from the methods in `client.rs`, instead the state of the `Client` is used.